cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/ExecutableBuilder.java

```diff
@@ -437,9 +438,20 @@ private WorkflowModellingResult compileWorkFlow(List<Map<String, Map<String, Obj
         return new WorkflowModellingResult(new Workflow(steps), errors);
     }
 
+    private String getStepName(Map<String, Map<String, Object>> stepRawData, List<RuntimeException> errors) {
+        String stepName = null;
+        try {
+            stepName = stepRawData.keySet().iterator().next();
+            executableValidator.validateStepName(stepName);
```

Comment:
getStepName has side effects. When calling the method two times with the same parameters makes the list populated twice with error message, a get method should be idempotent. Th way to go here is to have a metod validateStepName that validates a stepName and a separate getStepName method.